apps/src/templates/Dialog.jsx

```diff
@@ -138,8 +141,24 @@ const Dialog = React.createClass({
     onCancel: whenNoChildOfTypes(Buttons),
     confirmText: whenNoChildOfTypes(Buttons),
     onConfirm: whenNoChildOfTypes(Buttons),
+    confirmType: whenNoChildOfTypes(Buttons),
   }),
 
+  handleKeyDown(event) {
+    // Focus the next button when tab is pressed, to prevent the
+    // user from selecting elements outside of the dialog.
+    if (event.key === 'Tab') {
+      const buttons = this.refs.baseDialog.getButtons();
+      if (buttons.length) {
+        // Focus the next button, or the first button if none is focused.
+        const curIndex = buttons.findIndex(btn => btn === document.activeElement);
+        const nextIndex = (curIndex + 1) % buttons.length;
+        buttons[nextIndex].focus();
+      }
+      event.preventDefault();
```

Comment:
In theory it would. Do we have any dialogs that have input elements? If we're going to try to make this a general-purpose dialog component, we might want to use something like https://github.com/reactjs/react-modal which already tries to solve some of these problems